helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -279,12 +278,50 @@ public void subscribeStateChanges(final IZkStateListener listener) {
     }
   }
 
+  public void subscribeStateChanges(final org.I0Itec.zkclient.IZkStateListener listener) {
+    subscribeStateChanges(new IZkStateListener() {
```

Comment:
Still duplicated code. Let's have a single wrapper implement class for the IOItech interface.

Reply:
Moreover, mark the methods taking IOItech interface as deprecated. We are doing this purely for backward compatibility.